name.abuchen.portfolio/src/name/abuchen/portfolio/model/PortfolioTransaction.java

```diff
@@ -63,6 +64,75 @@ public String toString()
         }
     }
 
+    public static class PortfolioTransactionBuilder
+    {
+        private Type txType;
+        private LocalDateTime datetime;
+        private String currencyCode;
+        private long amount = 0;
+        private long shares = 0;
+        private Security security;
+        private long fees = 0;
+        private long taxes = 0;
+
+        public PortfolioTransactionBuilder(Type txType)
+        {
+            this.txType = txType;
+        }
+
+        public PortfolioTransactionBuilder transactionAt(LocalDateTime dateTime)
+        {
+            datetime = dateTime;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder withCurrency(String code)
+        {
+            currencyCode = code;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder withAmountOf(long amount)
+        {
+            this.amount = amount;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder numberOfShares(long shares)
+        {
+            this.shares = shares;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder forSecurity(Security security)
+        {
+            this.security = security;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder withCostsOf(long fees)
+        {
+            this.fees = fees;
+            return this;
+        }
+
+        public PortfolioTransactionBuilder withTaxAmountOf(long taxes)
+        {
+            this.taxes = taxes;
+            return this;
+        }
+
+        public PortfolioTransaction build()
+        {
+            Objects.requireNonNull(datetime, "datetime is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(currencyCode, "currencyCode is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(amount, "amount is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(security, "security is a required field"); //$NON-NLS-1$
+            Objects.requireNonNull(shares, "shares is a required field"); //$NON-NLS-1$
```

Comment:
Missing translation and capitalization

Reply:
Why? This exception should be part of the log file, not for the user.

Reply:
As above.